src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.cpp

```diff
@@ -752,10 +753,29 @@ void SolidMechanicsLagrangianFEM::crsApplyTractionBC( real64 const time,
         localIndex const numNodes = faceToNodeMap.sizeOfArray( kf );
         for( localIndex a=0; a<numNodes; ++a )
         {
-          localIndex const dof = blockLocalDofNumber[ faceToNodeMap( kf, a ) ] + component - dofRankOffset;
-          if( dof < 0 || dof >= localRhs.size() )
-            continue;
-          RAJA::atomicAdd< parallelDeviceAtomic >( &localRhs[ dof ], value * faceArea[kf] / numNodes );
+
+          // The flag component = 0, 1 and 2 are for the traction in the three orthogonal directions
+          // in the global cartesian coordinate system  
+          // The flag component = -1 is for the normal traction
```

Comment:
This is probably not the correct location where you should put your comment.
The value comes from `FieldSpecificationBase::getComponent` So this is where your comment should go.
E.g.: why didn't you copy/paste your comment below (line 796) too?

You should improve the user message [here](https://github.com/GEOSX/GEOSX/blob/b204d2c528d59533929bca7f7a06b4946b8b2da2/src/coreComponents/managers/FieldSpecification/FieldSpecificationBase.cpp#L48) too.

Reply:
Done!